### DISCUSSION TITLE
UDF NDTI index is different from default NDTI

### DISCUSSION THREAD

**Discussion by Azarozo19** [davidfrantz/force#329] [Q&A] [answered]
I am currently comparing the NDTI index from the UDF vs the default (No UDF). The parameters used for the default process are INDEX = NDTI, SENSORS = SEN2A SEN2B, DATE_RANGE = 2021-01-01 2021-12-31. The output is nDates = 22 with spectral and nodata values.
The UDF parameter file has the same settings. In the new function script the outarray output from the function forcepy_pixels has different amounts of bands per pixel and the values differ as well. Is it possible to debug it efficiently? Does someone have an idea why the nBands are so different?

`start = date.fromisoformat('1970-01-01')

def forcepy_init(dates, sensors, bandnames):
    """
    dates:     numpy.ndarray[nDates](int) days since epoch (1970-01-01)
    sensors:   numpy.ndarray[nDates](str)
    bandnames: numpy.ndarray[nBands](str)
    """

    return [f'{(start+timedelta(days=int(dat))).year}{(start+timedelta(days=int(dat))).month:02d}{(start+timedelta(days=int(dat))).day:02d}_{sens.decode("utf-8")}' for dat, sens in zip(dates, sensors)]

def forcepy_pixel(inarray, outarray, dates, sensors, bandnames, nodata, nproc):
    """
    inarray:   numpy.ndarray[nDates, nBands, nrows, ncols](Int16)
    outarray:  numpy.ndarray[nOutBands](Int16) initialized with no data values
    dates:     numpy.ndarray[nDates](int) days since epoch (1970-01-01)
    sensors:   numpy.ndarray[nDates](str)
    bandnames: numpy.ndarray[nBands](str)
    nodata:    int
    nproc:     number of allowed processes/threads
    Write results into outarray.
    """

    inarray = inarray.astype(np.float32)
    inarray = inarray[:, :, 0, 0]
    invalid = inarray == nodata  ## bool mask values

    valid = np.where(inarray[:, 0] != nodata)[0]  # skip no data; just check first band

    if len(valid) == 0:
        return
    inarray[invalid] = np.nan

    swir1 = np.argwhere(bandnames == b'SWIR1')[0][0]
    swir2 = np.argwhere(bandnames == b'SWIR2')[0][0]

    vals = inarray[valid, :]
   
    ndti = (vals[:, swir1] - vals[:, swir2]) / (vals[:, swir1] + vals[:, swir2])
    
    outarray[valid] = ndti*10000
`

**Comment by davidfrantz** [comment, marked as answer]
Dear @Azarozo19,

it is difficult to say without seeing the parameter files. I assume, though, that you used interpolation in the TSA submodule (default), whereas the UDF submodule has no such option.

Best,
David
